retry: avoid use of nil context

tbg opened the pull request:
Caught by Metacheck presumably because I'm running go1.9? Either way, this
was slightly sketchy.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17183)
<!-- Reviewable:end -->

tbg committed:
retry: avoid use of nil context

Caught by Metacheck presumably because I'm running go1.9? Either way, this
was slightly sketchy.

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17183#-:-KpoMPRCCGkvJHIcLAF3:bnfp4nl)*
<!-- Sent from Reviewable.io -->
